GOFevaluation/utils.py

```diff
@@ -136,7 +135,29 @@ def _get_count_density(ns, be_first, be_second, data_sample):
     return ns
 
 
-def _get_equiprobable_binning(reference_sample, n_partitions, order=None):
+def _weighted_equi(n_bins, data, weights):
+    """Perform a 1D equiprobable binning for the reference sample. The binning
+    is defined such that the number of counts in each bin are (almost) equal.
+    Bins are defined based on the ECDF of the reference sample.
+    :param n_bins: number of partitions in this dimension
+    :type n_bins: int
+    :param data: sample of unbinned reference
+    :type data: array_like, 1-Dimensional
+    :param weights: weights of samples
+    :type weights: array_like, 1-Dimensional
+    :return: Returns bin_edges.
+    :rtype: array_like, 1-Dimensional
+    """
+    weights = weights[data.argsort()]
+    data = data[data.argsort()]
+    cumsum = np.cumsum(weights)
+    cumsum -= cumsum[0]
+    bin_edges = np.interp(np.linspace(0, 1, n_bins + 1)[1:-1], cumsum / cumsum[-1], data)
+    bin_edges = np.hstack([-np.inf, bin_edges, np.inf])
```

Comment:
I guess definitely it can be faster with numba. But the outcome from that specified implementation is small because it is only a 1D equilbinning, so numpy is already fast enough for most cases.